jax/interpreters/pxla.py

```diff
@@ -452,8 +452,13 @@ def aval_to_result_handler(sharding_spec: Optional[ShardingSpec],
 PxlaResultHandler = Callable[..., Callable[[List[xb.xla_client._xla.PyLocalBuffer]], Any]]
 pxla_result_handlers: Dict[Type[core.AbstractValue], PxlaResultHandler] = {}
 pxla_result_handlers[core.AbstractUnit] = lambda *_: lambda _: core.unit
+
+def _array_result_handler(sharding_spec, indices, aval: ShapedArray, bufs):
+  return ShardedDeviceArray(aval, sharding_spec, bufs, indices)
+
 def array_result_handler(sharding_spec, indices, aval: ShapedArray):
-  return lambda bufs: ShardedDeviceArray(aval, sharding_spec, bufs, indices)
+  return partial(_array_result_handler, sharding_spec, indices, aval)
```

Comment:
I think I'm missing something -- what's the purpose of this extra indirection here? Like why partial over _array_result_handler instead of just the original code?

Reply:
Lambda cannot be introspected as easily as a `partial(x, ...)` where `partial(x, *args).args == args`